src/Sulu/Component/Content/Compat/CacheLifetimeBehaviourInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Compat;
+
+interface CacheLifetimeBehaviourInterface
```

Comment:
i guess it makes sense to extract this, but i am not sure about the name/namespace. this is different to the behaviours in `Sulu\Component\Content\Document\Behavior` (eg `AuthorBehavior`), so i am not sure if i would use `Behaviour` here. but i also dont have a better idea, so if everybody else is okay with the name, its okay for me too 🙈

Reply:
I don't wanna go with just `CacheLifetimeInterface`, but I guess `CacheLifetimeStructureInterface` would also work. wdyt? But this would imply, that the `CacheLifetimeStructureInterface` extends from `StructureInterface`, which it does not currently (but that wouldn't be a problem though)

Reply:
i would prefer `CacheLifetimeStructureInterface` to `CacheLifetimeBehaviourInterface` 🙂 but i would not extend the `StructureInterface`. i think its okay if it has just a single method

Reply:
There was also another Pull Request in the past targetting the same issue: https://github.com/sulu/sulu/pull/4893

Which did include `getController`, `getView` and `getCacheLifeTime` make more sense.

Reply:
@alexander-schranz What do you mean? Moving the `getCacheLifeTime` to the `StructureInterface` directly?

Reply:
That comment was from `danrot` not from myself. At the opening it would have been possible to add the methods to structureinterface (pre 2.x) but we did not do it as snippets don't need them. And at current state it would be a bc break so we can not add it to the structureinterface and I also think it is not a good idea. I more want to add to the discussion if the other 2 things `getController` and `getView` should be in the same interface as they always will live together. When we have a `controller` we will have a `view` and that response will have a `cachelifetime`.

Reply:
Okay but the ContentBundle does not have an interface yet defining the `getController()`, `getView()` and `getCacheLifeTime()` methods. So we could introduce that in sulu and implement it in the `ContentStructureBridge` too

Reply:
> So we could introduce that in sulu and implement it in the ContentStructureBridge too.

Yeah the content bundle and article bundle should then use the new interface. So in future they maybe work over the same controller, but atleast currently can use the same service setting and resolving the cachelifetime from the structure.

Currently the cachelifetime in the content bundle is just set on `_cacheLifeTime` attribute in the route default provider (which is very inconsistent to the others (page / articles)) but I think currently not read because it does not implement the `PageInterface`. So would introduce the interface and then we can adopt the bundles to go with that interface and do not have to check for method_exists longer.

What do the other think about Interface containing the 3 methods? /cc @wachterjohannes @niklasnatter

Reply:
so you would create a `RoutableStructureInterface` that contains `getController`, `getView` and `getCacheLifeTime`, right?
that sound good to me 🙂

Reply:
Did update the PR